docs/enhancements/error-logging-improvement.md

```diff
@@ -0,0 +1,62 @@
+# Error Logging Improvement
+
+## Table of Contents
+
+* [Summary](#summary)
+* [Objectives](#objectives)
+  * [Goals](#goals)
+  * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+  * [User Stories](#user-stories)
+* [Unresolved Questions](#unresolved-questions)
+
+# Summary
+
+The log produced by Sonobuoy is used to be able to diagnose clusters and
+troubleshoot users. This makes Sonobuoy's error logging a requirement, not a
+nice-to-have. A requirement must have automated testing or it's going to break.
+Error logging in Sonobuoy, at the time of writing this document, is difficult if
+not impossible to unit test.
+
+# Objectives
+
+The main objective is to unify how we write to the error log and treat the error
+log as a proper dependency.
+
+## Goals
+
+* Write unit tests for functions that call the error log.
```

Comment:
I don't mind writing unit tests on errors, but it's generally faux pas to write UT's on error logs.  Unless you have a utility to splice the error from the log.

Reply:
generally, I agree that it's a faux pas to test error logs. But this SEP outlines why we *should* test our error log calls which, to restate, is because we depend heavily on that *feature* and it needs to not have bugs like the one I squashed recently. The error log actually provides value to the consumer and therefore is a requirement.